cloud/misc/virt_net.py

```diff
@@ -206,6 +208,41 @@ def create(self, entryid):
             if not state:
                 return self.module.exit_json(changed=True)
 
+    def modify(self, entryid, xml):
+        if not self.module.check_mode:
+            network = self.find_entry(entryid)
+            # identify what type of entry is given in the xml
+            new_data = etree.fromstring(xml)
+            old_data = etree.fromstring(network.XMLDesc(0))
+            if new_data.tag == 'host':
+                mac_addr = new_data.get('mac')
+                hosts = old_data.xpath('/network/ip/dhcp/host')
+                # find the one mac we're looking for
+                host = None
+                for h in hosts :
+                    if h.get('mac') == mac_addr:
+                        host = h
+                        break
+                if host is None:
+                    # add the host
```

Comment:
After review the code again, I think the checkmode is not implemented the way it should.
IMHO it should be here and not on line 212, e.g.:
~~~python
if self.module.checkmode:
    return True
~~~